.github/workflows/build.yaml

```diff
@@ -46,10 +64,55 @@ jobs:
         PYTHON: python
     - name: Install dependencies
       run: make setup
+    - name: Create changelog and bump
+      if: ${{ github.ref == 'refs/heads/main' && matrix.os == env.ARTIFACT_OS && matrix.python == env.ARTIFACT_PYTHON }}
+      run: |
+        git config --global user.name "github-actions"
+        git config --global user.email "github-actions@github.com"
+        git config --list --global
+        # First generate the CHANGELOG diff to use as release notes.
+        mkdir -p ${{ env.ARTIFACT_DIR }}
+        cz changelog --dry-run --incremental --unreleased-version "Release Notes" > ${{ env.ARTIFACT_DIR }}/CHANGELOG.md
```

Comment:
The `mkdir` is needed because the `dist/` folder may not exist because `make dist` hasn’t run yet? I wonder if we should add that to our `make setup`?

That generated file `${{ env.ARTIFACT_DIR }}/CHANGELOG.md` isn’t the change log, though, it’s the release notes for this particular run, right?

Reply:
> The `mkdir` is needed because the `dist/` folder may not exist because `make dist` hasn’t run yet? I wonder if we should add that to our `make setup`?

Yes, that's correct. I'm happy to add it to `make setup` if you think it's better.

Reply:
> That generated file `${{ env.ARTIFACT_DIR }}/CHANGELOG.md` isn’t the change log, though, it’s the release notes for this particular run, right?

Yes, it's the changelog diff that we use as release notes.

Reply:
> I'm happy to add it to `make setup` if you think it's better.

Yes let’s do that. That’s right in line with issue https://github.com/jenstroeger/python-package-template/issues/262 as well 👍🏼